src/compiler/types.ts

```diff
@@ -498,6 +498,7 @@ namespace ts {
         /* @internal */ original?: Node;                // The original node if this is an updated node.
         /* @internal */ startsOnNewLine?: boolean;      // Whether a synthesized node should start on a new line (used by transforms).
         /* @internal */ jsDocComments?: JSDoc[];        // JSDoc for the node, if it has any.
+        /* @internal */ jsDocCache?: (JSDoc | JSDocParameterTag)[]; // JSDoc for the node, plus JSDoc retrieved from its parents

```

Comment:
Maybe add a comment explaining that `JSDocParameterTag` is treated differently because we want to avoid redoing the mapping from each tag to each parameter declaration.